internal/storage/authn/redis/store_test.go

```diff
@@ -0,0 +1,92 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	goredis "github.com/redis/go-redis/v9"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"go.flipt.io/flipt/internal/storage/authn"
+	authtesting "go.flipt.io/flipt/internal/storage/authn/testing"
+)
+
+type redisContainer struct {
+	testcontainers.Container
+	host string
+	port string
+}
+
+func setupRedis(ctx context.Context) (*redisContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        "redis:alpine",
+		ExposedPorts: []string{"6379/tcp"},
+		WaitingFor:   wait.ForLog("* Ready to accept connections"),
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	mappedPort, err := container.MappedPort(ctx, "6379")
+	if err != nil {
+		return nil, err
+	}
+
+	hostIP, err := container.Host(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &redisContainer{Container: container, host: hostIP, port: mappedPort.Port()}, nil
+}
+
+func TestAuthenticationStoreHarness(t *testing.T) {
```

Comment:
Take it or leave it:
```suggestion
func TestAuthenticationStoreRedis(t *testing.T) {
```

This is an instance of the test case invoking the harness to ensure specifically redis complies.